types/index.d.ts

```diff
@@ -0,0 +1,12 @@
+import {ChartType, Plugin} from 'chart.js';
+import {Options} from './options'
+
+declare module 'chart.js' {
+  interface PluginOptionsByType<TType extends ChartType> {
+    deferred?: Options
+  }
+}
+
+declare const deferred: Plugin;
+
+export default deferred;
```

Comment:
Currently, the plugin is [not exported](https://github.com/chartjs/chartjs-plugin-deferred/blob/master/src/plugin.js#L123). However, I would prefer to not auto-register it and export the plugin instead, as [it has been done](https://github.com/chartjs/chartjs-plugin-datalabels/commit/13daa478d9082fd0bc100e36394719215771d689) for the datalabels one. We can either merge this PR without the export, or submit another PR that removes the auto-registration of the plugin and then merge this one.

Reply:
Will work on that